### PR TITLE
Update client/swagger.yaml from 0.21.0 to b922c7d

### PR DESCRIPTION
**Pull request by kzys**
*Issue #, if available:*

NA

*Description of changes:*

Firecracker is making a few breaking changes. I'd like to estimate how much changes we may need, before their official release.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: machine.go [hunk 1]
```diff
@@ -613,9 +613,8 @@ func (m *Machine) setupLogging(ctx context.Context) error {
 	}
 
 	l := models.Logger{
-		LogFifo:       String(m.Cfg.LogFifo),
+		LogPath:       String(m.Cfg.LogFifo),
```
Comment (kzys) [inline machine.go:616]: It may be better to rename m.Cfg.LogFifo to LogPath as well.

**Comment by kzys** [comment]
Let me terminate the EC2 instance. It seems unhealthy. https://buildkite.com/firecracker-microvm/firecracker-go-sdk/builds/873

File: machine.go [hunk 2]
```diff
@@ -755,13 +761,14 @@ func (m *Machine) createNetworkInterface(ctx context.Context, iface NetworkInter
 	}
 
 	m.logger.Debugf("createNetworkInterface returned for %s", iface.StaticConfiguration.HostDevName)
+	m.networkInterfaces[ifaceID] = ifaceCfg
 	return err
 }
 
 // UpdateGuestNetworkInterfaceRateLimit modifies the specified network interface's rate limits
 func (m *Machine) UpdateGuestNetworkInterfaceRateLimit(ctx context.Context, ifaceID string, rateLimiters RateLimiterSet, opts ...PatchGuestNetworkInterfaceByIDOpt) error {
```
Comment (kzys) [inline machine.go:769]: The opts will be ignored on the PUT path. How about removing `opts`? Nobody is using the method, at least on GitHub.

https://github.com/search?q=UpdateGuestNetworkInterfaceRateLimit&type=Code

File: machine.go [hunk 3]
```diff
@@ -886,11 +911,21 @@ func (m *Machine) GetMetadata(ctx context.Context, v interface{}) error {
 // parameters of the partialDrive.
 func (m *Machine) UpdateGuestDrive(ctx context.Context, driveID, pathOnHost string, opts ...PatchGuestDriveByIDOpt) error {
```
Comment (kzys) [inline machine.go:912]: This `opts` will be ignored on the PUT path as well, but nobody is using that on GitHub.

https://github.com/search?q=UpdateGuestDrive&type=Code

File: machine.go (hunk 1 above)
Comment (kzys) [reply machine.go]: I've added LogPath and MetricsPath on c1286946e641dc4fc817e59c0f7455ed31d59942.

File: machine.go (hunk 2 above)
Comment (kzys) [reply machine.go:799]: @xibz @IRCody What do you think? I think removing that (while nobody is using, at least on GitHub) is okay.

Comment (IRCody) [reply machine.go:799]: If we remove it are we forcing breaking API changes in the future if FC adds an option here?

Comment (kzys) [reply machine.go:799]: It depends how we model their changes on the Go SDK. If Firecracker extends NetworkInterface / PartialNetworkInterface and adding opts makes sense from our perspective, then yes.

Another option is supporting only PatchGuestNetworkInterface. We need to update our tests and we might break other clients (technically Firecracker's change breaks the clients), but it is simpler and we are making less arbitrary decisions.

Comment (IRCody) [reply machine.go:799]: You mean only supporting PatchGuestNetworkInterface in UpdateGuestNetworkInterfaceRateLimit?

Comment (kzys) [reply machine.go:799]: Yes. And doing the same on UpdateGuestDrive()

Comment (kzys) [reply machine.go]: I've updated the implementation to only support PatchGuestNetworkInterface (post-boot configuration).

File: machine.go (hunk 3 above)
Comment (kzys) [reply machine.go]: I've reverted the change.

File: machine.go [hunk 4]
```diff
@@ -78,6 +78,9 @@ type Config struct {
 	// should be created.
 	SocketPath string
 
+	// LogPath defines the file path where the Firecracker log is located.
+	LogPath string
+
 	// LogFifo defines the file path where the Firecracker log named-pipe should
 	// be located.
 	LogFifo string
```
Comment (kzys) [inline machine.go:86]: @xibz @IRCody I'm supporting both LogPath (what Firecracker supports now) and LogFifo (what Firecracker supported before) here. What do you think? Would it be better to let clients break and keep our client thin?

Comment (IRCody) [reply machine.go:86]: I agree lets not break clients for this.

File: machine.go [hunk 5]
```diff
@@ -619,10 +626,32 @@ func (m *Machine) setupLogging(ctx context.Context) error {
...
-		m.Cfg.LogFifo,
-		m.Cfg.MetricsFifo,
-	)
+	m.logger.Debugf("Configured VMM logging to %s", path)
+
+	return nil
+}
+
+func (m *Machine) setupMetrics(ctx context.Context) error {
+	fmt.Println("setupMetrics")
```
Comment (IRCody) [inline machine.go:635]: m.logger.printf if we still want this message?

File: machine_test.go [hunk 6]
```diff
@@ -418,6 +418,60 @@ func TestStartVMM(t *testing.T) {
 	_, closed := <-m.exitCh
 	assert.False(t, closed)
 }
+func TestLogAndMetrics(t *testing.T) {
```
Comment (IRCody) [inline machine_test.go:421]: nit: newline between } and func here.

File: .buildkite/pipeline.yml [hunk 7]
```diff
@@ -85,7 +85,8 @@ steps:
       queue: "${BUILDKITE_AGENT_META_DATA_QUEUE:-default}"
       distro: "${BUILDKITE_AGENT_META_DATA_DISTRO}"
       hostname: "${BUILDKITE_AGENT_META_DATA_HOSTNAME}"
-
+    soft_fail:
```
[the remark this answers is not in the document]
Comment (kzys) [reply .buildkite/pipeline.yml:88]: After this change, this repos' master cannot be used with Firecracker 0.21.0. So we are going to allow non-zero exit status here.

File: machine.go (hunk 5 above)
Comment (kzys) [reply machine.go:635]: Right. Will fix.

File: machine_test.go (hunk 6 above)
Comment (kzys) [reply machine_test.go:421]: Will do.

**Pull request merged**